### PR TITLE
107 eeb lists

### LINKED ISSUE #107: Move EEB lists to seperate class

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #107 

More lists are removed now :)
The same holds here as for the ElementSelectionBehaviours:
All data is no longer divided over different classes, but is bundled into one data class.
There is now only a single implementation of the ElementEventBinding interface: GenericElementEventBinding. Instances of this class are created by the ElementEventBindingFactory.

Have fun reviewing :)

### CODE REVIEW

File: src/octopeer-github/content/ElementEventBinding/ElementEventBindingData.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+/// <reference path="../Types/EventID.ts"/>
+/**
+ * Created by Mathias on 02-06-2016.
+ * This interface enforces the data elements of the ElementEventBindings.
+ * eventID is the corresponding ID given in the eventID class.
+ * name is the descriptive string that belongs to the event.
```
Comment (mpsijm) [inline src/octopeer-github/content/ElementEventBinding/ElementEventBindingData.ts]: Please document the optional `hookToDOM` field here.

File: src/octopeer-github/content/ElementEventBinding/ElementEventBindingData.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    hookToDOM?: (elementSelectionBehaviour: GenericElementSelectionBehaviour) => void;
+    name: string;
+}
+
+/**
+ * elementEventBindingDataList contains all data about the different ElementEventBindings.
+ * @type {{eventID: EventID, name: string}[]}
+ */
+// tslint:disable-next-line:no-unused-variable
+const elementEventBindingDataList: ElementEventBindingData[] = [
```
Comment (mpsijm) [inline src/octopeer-github/content/ElementEventBinding/ElementEventBindingData.ts]: Please be consistent in your naming. I recall that this list for the ESBs was called `unsortedElementSelectionBehaviourData`, that doesn't quite match this one.

File: src/octopeer-github/content/ElementEventBinding/GenericElementEventBinding.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    public getEventType() {
+        return this.eventType;
+    }
+
+    /**
+     * Make sure the Event is hooked to the DOM.
+     * @param elementSelectionBehaviour The GenericElementSelectionBehaviour
+     * whose callback should be used on the firing of this Event.
+     */
+    private initDOMEvent(elementSelectionBehaviour: ElementSelectionBehaviour) {
```
Comment (mpsijm) [inline src/octopeer-github/content/ElementEventBinding/GenericElementEventBinding.ts]: How is the optional `hookToDOM` field used in the `GEEB`?

File: src/octopeer-github/main/Database/EventObject/EventFactory.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@
  * The utility class EventFactory can create the different EventObjects.
  * It seems redundant to have this, since TypeScript can infer an interface on any anonymous object.
  * However, so is stating the fieldnames of the objects every time a new EventObject is created.
- * These factory methods allow EventObjects to be instantiated based on an ordered tuple (parameter list) instead.
+ * These esbFactory methods allow EventObjects to be instantiated based on an ordered tuple (parameter list) instead.
```
Comment (mpsijm) [inline src/octopeer-github/main/Database/EventObject/EventFactory.ts]: Why exactly did you change this word? It's not about ESBs right?

File: src/octopeer-github/main/Options/DoNotWatchOptions.ts [hunk 5]
```diff
@@ -43,18 +40,18 @@ const DoNotWatchOptions = new (class DoNotWatchOptions {
...
         }
         if (Options.getDoNotWatchHoverEvents()) {
-            doNotWatchEvents.push(MouseEnterElementEventBinding);
-            doNotWatchEvents.push(MouseLeaveElementEventBinding);
+            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.MOUSE_ENTER));
+            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.MOUSE_LEAVE));
         }
         if (Options.getDoNotWatchKeyboardShortcutEvents()) {
-            doNotWatchEvents.push(KeystrokeElementEventBinding);
+            doNotWatchEvents.push(eebFactory.findElementEventBindingData(EventID.KEYSTROKE));
```
Comment (mpsijm) [inline src/octopeer-github/main/Options/DoNotWatchOptions.ts]: Again a note here, also for the Test of this class, I'm changing this in another branch. :)

File: src/octopeer-github/content/ElementEventBinding/ElementEventBindingFactory.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+/// <reference path="GenericElementEventBinding.ts"/>
+/**
+ * Created by Mathias on 2016-06-02.
+ */
+class ElementEventBindingFactory {
```
Comment (thervh70) [inline src/octopeer-github/content/ElementEventBinding/ElementEventBindingFactory.ts]: Missing Class documentation

File: src/octopeer-github/content/ElementEventBinding/GenericElementEventBinding.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
+/**
+ * Created by Mathias on 2016-06-02.
+ */
+class GenericElementEventBinding implements ElementEventBinding {
```
Comment (thervh70) [inline src/octopeer-github/content/ElementEventBinding/GenericElementEventBinding.ts]: Also missing documentation

File: src/octopeer-github/content/ElementEventBinding/GenericElementEventBinding.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    /**
+     * Construct a GenericElementEventBinding, assign the proper values to the field (from the data)
+     * and execute the initDOMEvent function
+     * @param elementSelectionBehaviour
+     * @param data
+     */
+    constructor(elementSelectionBehaviour: ElementSelectionBehaviour, data: ElementEventBindingData) {
+        this.eventID = data.eventID;
+        this.eventType = data.name;
+        if (data.hookToDOM !== undefined) {
```
Comment (mdingjan) [inline src/octopeer-github/content/ElementEventBinding/GenericElementEventBinding.ts]: Can you reach the code in this if-statement with a test case?

**Pull request merged**